flink-staging/flink-ml/src/main/scala/org/apache/flink/ml/evaluation/Scorer.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.ml.evaluation
+
+import org.apache.flink.api.scala._
+import org.apache.flink.ml.common.{FlinkMLTools, ParameterMap, WithParameters}
+import org.apache.flink.ml.pipeline.Predictor
+
+import org.apache.flink.ml.pipeline.EvaluateDataSetOperation
+
+//TODO: Need to generalize type of Score (and evaluateOperation)
+class Scorer(val metric: Score[Double]) extends WithParameters {
+
+  def evaluate[Testing, PredictorInstance<: Predictor[PredictorInstance]](
+      testing: DataSet[Testing],
+      predictorInstance: PredictorInstance,
+      evaluateParameters: ParameterMap = ParameterMap.Empty)
+      (implicit evaluateOperation: EvaluateDataSetOperation[PredictorInstance, Testing, Double]):
+    DataSet[Double] = {
+
+    FlinkMLTools.registerFlinkMLTypes(testing.getExecutionEnvironment)
```

Comment:
True, I will remove this.

BTW, the we really need to find some different names for the functions, we shouldn't have 2  functions named `evaluate` that are so related, but do different things.